client/state/sites/sharing-buttons/actions.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	SHARING_BUTTONS_RECEIVE,
+	SHARING_BUTTONS_REQUEST,
+	SHARING_BUTTONS_REQUEST_FAILURE,
+	SHARING_BUTTONS_REQUEST_SUCCESS,
+	SHARING_BUTTONS_SAVE,
+	SHARING_BUTTONS_SAVE_FAILURE,
+	SHARING_BUTTONS_SAVE_SUCCESS,
+	SHARING_BUTTONS_UPDATE
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that sharing buttons have been received.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The sharing buttons object
+ * @return {Object}        Action object
+ */
+export function receiveSharingButtons( siteId, settings ) {
+	return {
+		type: SHARING_BUTTONS_RECEIVE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action object to be used in signalling that some sharing buttons have been updated.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} settings The updated sharing buttons
+ * @return {Object}        Action object
+ */
+export function updateSharingButtons( siteId, settings ) {
+	return {
+		type: SHARING_BUTTONS_UPDATE,
+		siteId,
+		settings
+	};
+}
+
+/**
+ * Returns an action thunk which, when invoked, triggers a network request to
+ * retrieve sharing buttons
+ *
+ * @param  {Number} siteId Site ID
+ * @return {Function}      Action thunk
+ */
+export function requestSharingButtons( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SHARING_BUTTONS_REQUEST,
+			siteId
+		} );
+
+		return wpcom.undocumented().sharingButtons( siteId )
```

Comment:
Had you considered using the API middleware? One other advantage I see in doing so is being able to control the "is already requesting? then skip" logic that we currently maintain in the query component.